lumigator/python/mzai/backend/backend/api/routes/jobs.py

```diff
@@ -129,21 +129,67 @@ def list_jobs(
 
 
 @router.get("/{job_id}")
-def get_job(service: JobServiceDep, job_id: UUID) -> Job:
-    """Attempts to retrieve merged job data from the Lumigator repository and Ray.
+def get_job(
+    service: JobServiceDep,
+    job_id: UUID,
+) -> Job:
+    """Attempts to retrieve merged job data from the Lumigator repository and Ray
+    for a valid UUID.
+
+    The result is a merged representation which forms an augmented view of a 'job'.
 
     NOTE: Lumigator repository data takes precedence over Ray metadata.
     """
     job = service.get_job(job_id)
     ray_job = _get_ray_job(job_id)
 
     # Combine both types of response.
-    x = ray_job.model_dump()  # JobSubmissionResponse
-    y = job.model_dump()  # JobResponse
-    merged = {**x, **y}
+    ray_info = ray_job.dict()
+    lm_info = job.model_dump()
+    merged = {**ray_info, **lm_info}
     return Job(**merged)
 
 
+@router.get("/{job_type}/")
```

Comment:
This seems slightly strange since already have a get route for /{job_id}. Is there a better way to make it clear that one route is getting a job, and another list listing jobs per type? It seems like this works, but may confuse people who are exploring our swagger docs/etc, not sure.

Reply:
Please take a read at https://github.com/mozilla-ai/lumigator/pull/576#discussion_r1907515466 for reference. TL;DR any choice seems bad 😅 This was chosen to keep in line with the current POSTs and avoid overloading the root jobs route. But maybe some additional comments in the openapi spec could do the trick; I'll inspect the generated spec and see how it looks.

Reply:
IMO I don't think a separate endpoint is needed, how about a query param for filtering?

i.e GET /jobs <-- returns all jobs
GET /jobs?job_type=evaluate <-- returns only evaluate jobs

Reply:
LGTM!!!! @aittalam @veekaybee what do you think?
`GET /jobs?job_types=evaluate` would seem ok to me (and iirc this _may_ allow unifying some calls in the underlying service/records as a bonus)

Reply:
I love @khaledosman's suggestion! My vote is for that 💯